binary_search_tree/binary_search_tree.py

```diff
@@ -1,14 +1,45 @@
 class BinarySearchTree:
-  def __init__(self, value):
-    self.value = value
-    self.left = None
-    self.right = None
+    def __init__(self, value):
+        self.value = value
+        self.left = None
+        self.right = None
 
-  def insert(self, value):
-    pass
+    # this implementation doesn't balance itself or allow duplicates
+    def insert(self, value):
+        if value < self.value:
```

Comment:
for practice purposes try to make your insert implementation a self balancing. Good job on your tree max!